src/context.rs

```diff
@@ -110,18 +110,18 @@ impl Context {
   #[inline(always)]
   pub fn set_promise_hooks(
     &self,
```

Comment:
A problem not introduced by your PR, however for posterity:
An API that mutates the `Isolate` or any object on the V8 heap should have at least one `&mut` parameter. 
Usually that would be an `&mut HandleScope` but if that's not possible somehow there must be something else.